### PR TITLE
Add support for SSO flow

### PR DESCRIPTION
**Pull request by Rizziepit**

### CODE REVIEW

**Comment by Rizziepit** [comment]
@miltontony @smn ready for review

File: unicore/hub/client/__init__.py [hunk 1]
```diff
@@ -52,8 +56,41 @@ def put(self, path, *args, **kwargs):
...
+
+        # check that url is absolute
+        parts = urlparse(login_callback_url)
+        if not (parts.scheme and parts.netloc):
+            raise ValueError('login_callback_url must be absolute')
+
+        return login_callback_url
+
+    def get_login_redirect_url(self, login_callback_url=None):
+        # TODO: enforce https
```
Comment (miltontony) [inline unicore/hub/client/__init__.py]: Is this to ensure we use `https://hub.unicore.io`?

Comment (Rizziepit) [inline unicore/hub/client/__init__.py]: Yes. Will we install certs for QA as well?

**Comment by miltontony** [comment]
:+1:

**Comment by Rizziepit** [comment]
@miltontony can you review https://github.com/universalcore/unicore.hub.client/commit/996f2fcf70007afcf11e2781b0d10a115be07e46

**Comment by miltontony** [comment]
:+1:

**Pull request merged**